name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVImporter.java

```diff
@@ -457,45 +463,70 @@ public void setFirstLineHeader(boolean isFirstLineHeader)
-                            CSVStrategy.ESCAPE_DISABLED, false, false, false, false);
+        CSVStrategy strategy = new CSVStrategy(delimiter, '"', CSVStrategy.COMMENTS_DISABLED,
+                        CSVStrategy.ESCAPE_DISABLED, false, false, false, false);
 
-            CSVParser parser = new CSVParser(reader, strategy);
+        CSVParser parser = new CSVParser(reader, strategy);
 
-            for (int ii = 0; ii < skipLines; ii++)
-                parser.getLine();
+        for (int ii = 0; ii < skipLines; ii++)
+            parser.getLine();
 
-            List<String[]> input = new ArrayList<>();
-            String[] header = null;
-            String[] line = parser.getLine();
-            if (isFirstLineHeader)
-            {
-                header = line;
-            }
-            else
-            {
-                header = new String[line.length];
-                for (int ii = 0; ii < header.length; ii++)
-                    header[ii] = MessageFormat.format(Messages.CSVImportGenericColumnLabel, ii + 1);
-                input.add(line);
-            }
+        List<String[]> input = new ArrayList<>();
+        String[] header = null;
+        String[] line = parser.getLine();
+        if (isFirstLineHeader)
+        {
+            header = line;
+        }
+        else
+        {
+            header = new String[line.length];
+            for (int ii = 0; ii < header.length; ii++)
+                header[ii] = MessageFormat.format(Messages.CSVImportGenericColumnLabel, ii + 1);
+            input.add(line);
+        }
 
-            while ((line = parser.getLine()) != null)
-                input.add(line);
+        while ((line = parser.getLine()) != null)
+            input.add(line);
 
-            this.columns = new CSVImporter.Column[header.length];
-            for (int ii = 0; ii < header.length; ii++)
-                this.columns[ii] = new Column(ii, header[ii]);
+        this.columns = new CSVImporter.Column[header.length];
+        for (int ii = 0; ii < header.length; ii++)
+            this.columns[ii] = new Column(ii, header[ii]);
 
-            this.values = input;
+        this.values = input;
 
-            mapToImportDefinition();
+        mapToImportDefinition();
+    }
+
+    public void processFile() throws IOException
+    {
+        IOException buffer = null;
+        try (FileInputStream stream = new FileInputStream(inputFile))
+        {
+            processStream(stream);
+            return;
+        }
+        catch (IOException e)
+        {
+            buffer = e;
+        }
+        byte[] ptext = inputFile.toString().getBytes(StandardCharsets.UTF_8);
+        String str = new String(ptext, StandardCharsets.ISO_8859_1);
+        Path path = Paths.get(URI.create("file://" + str)); //$NON-NLS-1$
+        try (InputStream stream = Files.newInputStream(path))
+        {
+            processStream(stream);
+        }
+        catch (IOException e)
+        {
+            if (buffer != null)
```

Comment:
BTW, my IDE says here *Redundant null check: The variable buffer cannot be null at this location*. Which I agree too. I will change the code accordingly.

Reply:
Yep. This is an orphan. While working in it I wanted to get the initial exception... But it is useless now...